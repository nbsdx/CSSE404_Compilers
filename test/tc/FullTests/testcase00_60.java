/*
 * Testcase designed to test lexical scope shadowing.
 * This should not pass, and the compiler should give an error to the effect 
 * of "variable at lexical scope shadowed by new definition".
 */

class Test3 {
	public static void main(String[] args) {
		int x = 3;
		while(x < 10) {
			int x = 5;
			x = x + 1;
		}
	}
}

// vim:filetype=java:textwidth=80
