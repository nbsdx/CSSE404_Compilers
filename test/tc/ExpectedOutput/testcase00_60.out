The variable x is already declared in the current scope.
